File: ruby-gem/lib/models/pick_up_step.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #PickUpStep
require_relative '../core/semantic_object'
require_relative 'step'


module DfcLinkmlConnector
  module Models
    class PickUpStep < Step
      SEMANTIC_TYPE = "dfc-b:PickUpStep".freeze

    end
  end
end

File: typescript-connector/src/models/PickUpStep.ts
// Class from DFC Business Ontology: #PickUpStep
import { SemanticObject } from "../core/SemanticObject.js";
import { Step, type StepParams } from "./Step.js";

export interface PickUpStepParams extends StepParams {}
export class PickUpStep extends Step {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:PickUpStep";
  }



  constructor(
    semanticId: string,
    params?: PickUpStepParams,
  ) {
    super(semanticId, params);
    this.semanticType = PickUpStep.SEMANTIC_TYPE;
  }
  static {
    SemanticObject.typeRegistry.set(PickUpStep.SEMANTIC_TYPE, PickUpStep);
  }
}
